### PR TITLE
Ability to more flexibly define the path to a text file

### PR DESCRIPTION
**Pull request by dputler**
This allows a user to use tilde expansion or simply the file name if the text file is in the user working directory.

### CODE REVIEW

File: pkg/R/context.R [hunk 1]
```diff
@@ -19,6 +19,10 @@
 #'}
 
 textFile <- function(sc, path, minSplits = NULL) {
+  # Allow the user to have a more flexible definiton of the text file path
+  path <- path.expand(path)
+  if (!(grepl("\\\\", path) | grepl("/", path)))
```
Comment (dputler) [inline pkg/R/context.R]: Hi Shivaram,

 

I’m fine with removing the lines of code that pre-pend the current working directory and going with the path expansion. The one difference in what you and I have done is that I ran:

 

sudo R CMD INSTALL pkg

 

from within the SparkR-pkg directory, so the SparkR package is the main R library. At this point, after a partial, unsuccessful, sync of my fork with the upstream copy, my local copy is a mess, so I can’t test whether that is causing the difference in what you and I found.

 

Should I retract this pull request, and submit a new one, or is this a change you can make?

 

Dan

Comment (shivaram) [inline pkg/R/context.R]: @dputler - No worries about the git client. I can fix things up during merge. One last thing I wanted to check with you was whether using `normalizePath` is better than using `path.expand`. From the documentation it looks like `normalizePath` calls `path.expand` first and then also handles things like `../` etc.

Comment (dputler) [inline pkg/R/context.R]: HI Shivaram,

 

normalizePath does look like a better choice for this.

 

Dan

Comment (shivaram) [inline pkg/R/context.R]: Cool. I'll make the switch from path.expand to normalizePath during the merge.

**Comment by shivaram** [comment]
Thanks @dputler -- I merged this manually with the changes discussed.

**Pull request merged**